src/kfmon.h

```diff
@@ -68,14 +68,55 @@ typedef struct {
 // (e.g., a pointer to a kfmon_watch_list_t, or NULL if no storage is needed).
 typedef int (*ipc_handler_t)(int, void *);
 
+// Free all resources allocated by a list and its nodes
```

Comment:
I'd rather keep code out of header files (unless it is a private helper which needs to be shared between multiple files), as I don't currently track them in the Makefile.